lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -2753,27 +2820,31 @@ void addAnnotationDecorations(SPIRVEntry *E, DecorationsInfoVec &Decorations) {
 
     switch (I.first) {
     case DecorationUserSemantic:
-      E->addDecorate(new SPIRVDecorateUserSemanticAttr(E, I.second));
+      assert(I.second.size() == 1);
```

Comment:
Maybe it'd be better to use `checkError` here instead of assertions to fail in a more gentle way. However, it's a recommendation, not a request.